### DISCUSSION TITLE
kong available plugins on server list

### DISCUSSION THREAD

**Discussion by naolaregaqena** [Kong/kong#8996] [Help]
On the kong admin api, from the `http://{kong-host}:8001/` endpoints response. `available_on_server` doesn't include custom added plugin. but it shows in the `loaded_plugins` property. it there a way to make it available there. konga apparently gets its plugin lists from there.

**Comment by naolaregaqena** [comment]
hello, any help here?

**Comment by flrgh** [comment]
Hello @naol-arega :wave: 

Can you please include the output from the admin API so that we can take a look? Also, it would be helpful to know which version of Kong you are running as well as if there are any error messages in the error.log file.

**Reply by naolaregaqena** [reply]
kong/kong-gateway: 2.8-alpine
postgres: 14.3-alpine
docker: 20.10.17
this is the output of `loaded_plugins`
```
"loaded_plugins": {
      "application-registration": true,
      "oauth2-introspection": true,
      "proxy-cache-advanced": true,
      "openid-connect": true,
      "forward-proxy": true,
      "canary": true,
      "request-transformer-advanced": true,
      "response-transformer-advanced": true,
      "rate-limiting-advanced": true,
      "ldap-auth-advanced": true,
      "statsd-advanced": true,
      "route-by-header": true,
      "jwt-signer": true,
      "vault-auth": true,
      "collector": true,
      "request-validator": true,
      "mtls-auth": true,
      "graphql-proxy-cache-advanced": true,
      "graphql-rate-limiting-advanced": true,
      "degraphql": true,
      "route-transformer-advanced": true,
      "kafka-log": true,
      "kafka-upstream": true,
      "exit-transformer": true,
      "key-auth-enc": true,
      "upstream-timeout": true,
      "mocking": true,
      "opa": true,
      "jq": true,
      "oauth2-token-introspection-request": true,
      "jwt": true,
      "acl": true,
      "correlation-id": true,
      "cors": true,
      "oauth2": true,
      "tcp-log": true,
      "udp-log": true,
      "file-log": true,
      "http-log": true,
      "key-auth": true,
      "hmac-auth": true,
      "basic-auth": true,
      "ip-restriction": true,
      "request-transformer": true,
      "response-transformer": true,
      "request-size-limiting": true,
      "rate-limiting": true,
      "response-ratelimiting": true,
      "syslog": true,
      "loggly": true,
      "datadog": true,
      "ldap-auth": true,
      "statsd": true,
      "bot-detection": true,
      "aws-lambda": true,
      "request-termination": true,
      "prometheus": true,
      "proxy-cache": true,
      "session": true,
      "acme": true,
      "grpc-gateway": true,
      "grpc-web": true,
      "pre-function": true,
      "post-function": true,
      "azure-functions": true,
      "zipkin": true
    }
```
and `available_on_server`
```
"available_on_server": {
      "jwt": true,
      "acl": true,
      "correlation-id": true,
      "cors": true,
      "oauth2": true,
      "tcp-log": true,
      "udp-log": true,
      "file-log": true,
      "http-log": true,
      "key-auth": true,
      "hmac-auth": true,
      "basic-auth": true,
      "ip-restriction": true,
      "request-transformer": true,
      "response-transformer": true,
      "request-size-limiting": true,
      "rate-limiting": true,
      "response-ratelimiting": true,
      "syslog": true,
      "loggly": true,
      "datadog": true,
      "ldap-auth": true,
      "statsd": true,
      "bot-detection": true,
      "aws-lambda": true,
      "request-termination": true,
      "prometheus": true,
      "proxy-cache": true,
      "session": true,
      "acme": true,
      "grpc-gateway": true,
      "grpc-web": true,
      "pre-function": true,
      "post-function": true,
      "azure-functions": true,
      "zipkin": true
    }
```
`oauth2-token-introspection-request` is missing on `available_on_server` and there is no error in the error log.

**Reply by naolaregaqena** [reply]
here is the entire response if it helps. i had to zip it. github doesn't support json file.
[index.json.zip](https://github.com/Kong/kong/files/8998268/index.json.zip)

**Reply by flrgh** [reply]
@naol-arega it appears you are running Kong enterprise (either in [free mode](https://docs.konghq.com/gateway/latest/#packages-and-modes) or without a current license), as denoted by `"version": "2.8.1.1-enterprise-edition"` in the API response.

The output of `plugins.available_on_server` is handled a little differently in this mode due to the need to account for which plugins are bundled with OSS vs plugins that are only available in full enterprise mode. This is why the API output also contains `plugins.disabled_on_server`. I see that it does not take 3rd party plugins into account when calculating the API response; I'll file an issue to investigate this more internally. If it is causing usability problems for you at the moment, it would be good to contact your sales rep or customer success engineer for assistance so that it can be prioritized properly.

Lastly, if your intention was to run the OSS version of Kong and _not_ Kong enterprise, you can find the OSS image at the [official kong image repository](https://hub.docker.com/_/kong/). Simply replace `kong/kong-gateway:2.8-alpine` with `kong:2.8` (alpine is the default image flavor) wherever you are specifying the docker image in your tooling.

**Reply by naolaregaqena** [reply]
i installed it based on the guide on the konghq docs [kong docker](https://docs.konghq.com/gateway/latest/install-and-run/docker/). i tried `kong:2.8.1` image but it has issue with openssl and other related things. to resolve those issue i tried installing `openssl` luarocks to the image, but it requires `lua 5.1` or greater. so i am going to stick with `kong/kong-gateway:2.8-alpine`. nevertheless third party plugins should be included in `available_on_server` .

**Reply by Tieske** [reply]
@naol-arega you might want to file an issue with Konga as well, as they probably want to use the other list. Especially with some changes coming up in 3.0, which will expose plugin version and priorities; see https://github.com/Kong/kong/pull/8821/files

**Reply by naolaregaqena** [reply]
ok i will. and also from what i saw in the code base, it is including version and priorities in the api. and `available_on_server` is a copy of `loaded_plugins`. just to make sure though i might file issue. thank you for the help.
